keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -172,39 +173,89 @@ Kubelet components.
 
 If the hints are not compatible, the Topology Manager may choose to
 reject the pod. Behavior in this case depends on a new Kubelet configuration
-value to choose the topology policy. The Topology Manager supports two
-modes: `strict` and `preferred` (default). In `strict` mode, the pod is
-rejected if alignment cannot be satisfied. The Topology Manager could
-use `softAdmitHandler` to keep the pod in `Pending` state.
+value to choose the topology policy. The Topology Manager supports four
+policies: `none`(default), `best-effort`, `restricted` and `single-numa-node`. 
+
+#### Policies
+
+**none (default)**: Kubelet does not consult Topology Manager for placement decisions 
+
+**best-effort**: Topology Manager will provide the optimal allocation for the container based
+on the hints provided by the Hint Providers, in most cases this will be the narrowest set of NUMA Node(s)
+that the request can be satisfied  on. If a non-optimal allocation is determined, the pod will be admitted
+with this non-optimal allocation.
+
+**restricted**:  Topology Manager will provide the optimal allocation for the container based
+on the hints provided by the Hint Providers. If a non-optimal allocation is determined, the pod will be rejected. 
```

Comment:
I think we need to expand more on what “optimal” vs “non-optimal” means concretely here. This point was underspecified in the original draft as well and it took us a lot of cycles to figure out that heuristic. Additionally, we call this decision and field “preferred” elsewhere. So perhaps we could s/optimal/preferred for simplicity.

Maybe we can add something like:

> A preferred alignment is defined as follows:
> * For each topology-aware resource manager, there is a possible assignment that satisfies the request, such that the least possible number of NUMA nodes is involved (calculated as if the node were empty.)
> * There is a possible assignment where the union of involved NUMA nodes for all such resources is no larger than the width required for any single resource.

Reply:
Moved the topology hint explanation to above this section, then added in an explanation of preferred and removed optimal. Take a look and see if its okay!